### PR TITLE
Add project CRUD to gRPC API

### PR DESCRIPTION
**Pull request by furuholm**
Replaces PR #81. Squashed history to make it easier to review.

### CODE REVIEW

File: samples/server/go-server/api/server/storage/memstore.go [hunk 1]
```diff
@@ -28,21 +29,74 @@ import (
...
+}
+
+// ListProjects returns the project id for all projects from the mem store
+func (m *memStore) ListProjects(filters string) []*pb.Project {
+	m.RLock()
+	defer m.RUnlock()
+	projects := make([]*pb.Project, len(m.projects))
+	i := 0
+	for k := range m.projects {
+		projects[i] = &pb.Project{Name: name.FormatProject(k)}
```
Comment (skelterjohn) [inline samples/server/go-server/api/server/storage/memstore.go]: `i` and `k` always have the same values here.

Comment (furuholm) [reply samples/server/go-server/api/server/storage/memstore.go]: I think that would be the case if m.Projects was an array, but m.Projects is a `map[string]bool` (to emulate a set).

Comment (skelterjohn) [reply samples/server/go-server/api/server/storage/memstore.go]: whoops, sorry. saw `projects :=` right above and I lost track of which was which.

**Comment by R2wenD2** [comment]
@furuholm I'll merge this tomorrow morning if you update/merge conflicts.

**Comment by furuholm** [comment]
@R2wenD2 Thanks! I have rebased on top of master.

**Pull request merged**